tests/subsys/mgmt/mcumgr/os_mgmt_info/src/smp_test_util.c

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "smp_test_util.h"
+#include <os_mgmt/os_mgmt.h>
+#include <zephyr/net/buf.h>
+#include <zcbor_encode.h>
+
+/* SMP header function for generating os_mgmt info command header with sequence number set to 1 */
+void smp_make_hdr(struct mgmt_hdr *rsp_hdr, size_t len)
+{
+	*rsp_hdr = (struct mgmt_hdr) {
+		.nh_len = len,
+		.nh_flags = 0,
+		.nh_op = 0,
+		.nh_group = MGMT_GROUP_ID_OS,
+		.nh_seq = 1,
+		.nh_id = OS_MGMT_ID_INFO,
+	};
+
+	mgmt_hton_hdr(rsp_hdr);
```

Comment:
This is internal function of SMP(rotocol) processing that should have never been exported, but mgmt interface is a mess so it was. It is even named wrongly.
This test implements client so it can spare some additional line to implement the client side of header encoding.

Reply:
Fixed